src/api/filesystem.go

```diff
@@ -0,0 +1,143 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	wh "github.com/skycoin/skycoin/src/util/http"
+)
+
+// saveDataRequest
+type saveDataRequest struct {
+	Data   map[string]interface{} `json:"data"`
+	Update bool                   `json:"update"`
+}
+
+// Save arbitrary data to disk
+// URI: /api/v2/data
+// Method: POST
+// Args:
+//     data: arbitrary data to save [required]
+//     update: update existing values [optional]
+func dataSaveHandler(gateway Gatewayer, w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		wh.Error405(w)
+		return
+	}
+
+	if r.Header.Get("Content-Type") != ContentTypeJSON {
+		wh.Error415(w)
+		return
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	decoder.DisallowUnknownFields()
+
+	var params saveDataRequest
+	err := decoder.Decode(&params)
+	if err != nil {
+		logger.WithError(err).Error("invalid save data request")
+		wh.Error400(w, err.Error())
+		return
+	}
+	defer r.Body.Close()
+
+	if params.Data == nil {
+		wh.Error400(w, "empty data")
+		return
+	}
+
+	err = gateway.SaveData(params.Data, params.Update)
+	if err != nil {
+		resp := NewHTTPErrorResponse(http.StatusInternalServerError, err.Error())
+		writeHTTPResponse(w, resp)
+		return
+	}
+
+	wh.SendJSONOr500(logger, w, "success")
```

Comment:
Wrong type of response for v2 api.
No need to include "success" message; leave it empty. The `200 OK` is sufficient to indicate success